sdcm/cluster.py

```diff
@@ -417,6 +417,33 @@ def wait_cs_installed(self, verbose=True):
         wait.wait_for(func=self.cs_installed, step=60,
                       text=text)
 
+    def update_db_binary(self, new_scylla_bin=None):
+        if new_scylla_bin:
+            self.log.info('Upgrading a DB binary for Node started')
+
+            self.remoter.send_files(new_scylla_bin, '/tmp/scylla', verbose=True)
+
+            # TODO: We should wait for scylla-server.service and scylla-ami-setup.service
+            # has finished instead of sleep here.
+            # As a workaround, sleep 3 minutes hoping they will finish
+            # With --stop-services option in ami, /usr/lib/scylla/scylla_prepare
+            # will fail the scylla-server.service
+            time.sleep(180)
+
+            # stop scylla-server before replace the binary
+            self.remoter.run('sudo systemctl stop scylla-server.service')
+
+            # replace the binary
+            self.remoter.run('sudo cp -f /usr/bin/scylla /usr/bin/scylla.origin')
+            self.remoter.run('sudo cp -f /tmp/scylla /usr/bin/scylla')
+            self.remoter.run('sudo chown root.root /usr/bin/scylla')
+            self.remoter.run('sudo chmod +x  /usr/bin/scylla')
+
+            self.remoter.run('sudo systemctl restart scylla-server.service')
+            self.remoter.run('sudo systemctl restart scylla-jmx.service')
```

Comment:
On Jun 8, 2016 6:47 AM, "Amos Jianjun Kong" <notifications@github.com>
wrote:
>
> In sdcm/cluster.py:
>
> > +            # As a workaround, sleep 3 minutes hoping they will finish
> > +            # With --stop-services option in ami,
/usr/lib/scylla/scylla_prepare
> > +            # will fail the scylla-server.service
> > +            time.sleep(180)
> > +
> > +            # stop scylla-server before replace the binary
> > +            self.remoter.run('sudo systemctl stop
scylla-server.service')
> > +
> > +            # replace the binary
> > +            self.remoter.run('sudo cp -f /usr/bin/scylla
/usr/bin/scylla.origin')
> > +            self.remoter.run('sudo cp -f /tmp/scylla /usr/bin/scylla')
> > +            self.remoter.run('sudo chown root.root /usr/bin/scylla')
> > +            self.remoter.run('sudo chmod +x  /usr/bin/scylla')
> > +
> > +            self.remoter.run('sudo systemctl restart
scylla-server.service')
> > +            self.remoter.run('sudo systemctl restart
scylla-jmx.service')
>
> Do we need to recover the update if the new binary fails to start?

No need. If the new binary fails, the test fails. We want to update the
binary because we want to test the new binary.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.